brave/src/main/java/brave/Tracing.java

```diff
@@ -145,7 +145,7 @@ public final Clock clock(TraceContext context) {
     boolean traceId128Bit = false, supportsJoin = true;
     Propagation.Factory propagationFactory = B3Propagation.FACTORY;
     ErrorParser errorParser = new ErrorParser();
-    List<FinishedSpanHandler> finishedSpanHandlers = new ArrayList<>();
+    Set<FinishedSpanHandler> finishedSpanHandlers = new LinkedHashSet<>(); // dupes not ok
```

Comment:
I think I'd expect an `add` API like this one to accept duplicates. While there probably are very few use cases, if seems more natural.

Reply:
this intentionally prevents it. The issue is that in spring we may have implemented multiple interfaces and do not want to accidentally report data twice. I would be happy to hear complaints vs the otherwise.

Reply:
PS all that's needed if one really really wants to send the same data with a function that has the same identity, is to wrap it. When someone comes up with that use case, point the anger at me :P